File: api-dot-net/Services/CodeExecution/Store/ExecutionJobStore.cs

using System.Collections.Concurrent;
using CjsApi.Services.CodeExecution.Dto;
namespace CjsApi.Services.CodeExecution.Store
{


    public class ExecutionJobStore
    {
        private readonly ConcurrentDictionary<string, ExecutionJob> _jobs = new();

        public ExecutionJob CreateJob()
        {
            var job = new ExecutionJob();
            _jobs[job.JobId] = job;
            return job;
        }

        public ExecutionJob? GetJob(string jobId)
            => _jobs.TryGetValue(jobId, out var job) ? job : null;
    }

}

File: java-api/cjs-service/src/main/java/com/cjs/cjs_service/service/codeExecutionSerivce/store/ExecutionJobStore.java
package com.cjs.cjs_service.service.codeExecutionSerivce.store;

import org.springframework.stereotype.Component;

import com.cjs.cjs_service.dto.ExecutionJob;

import java.util.concurrent.ConcurrentHashMap;

@Component
public class ExecutionJobStore {

    private final ConcurrentHashMap<String, ExecutionJob> jobs = new ConcurrentHashMap<>();

    public ExecutionJob createJob() {
        return createJob(-1);
    }

    public ExecutionJob createJob(int submissionId) {
        ExecutionJob job = new ExecutionJob();
        if (submissionId != -1) {
            job.setSubmissionId(submissionId);
        }
        jobs.put(job.getJobId(), job);
        return job;
    }

    public ExecutionJob getJob(String jobId) {
        return jobs.get(jobId);
    }
}
